cmd/cluster/health.go

```diff
@@ -85,101 +102,158 @@ func (o *healthOptions) run() error {
+			clusterHealthClient, err = osdCloud.NewAwsCluster(ocmClient, o.clusterID, o.awsProfile)
+			if err != nil {
+				return err
+			}
+			ownedLabel = "kubernetes.io/cluster/" + infraID
+			defer clusterHealthClient.Close()
+		} else {
+			return errors.New(fmt.Sprintf("Unknown cloud provider found: %s", cluster.CloudProvider().ID()))
+		}
+		err = clusterHealthClient.Login()
 		if err != nil {
 			return err
 		}
-		ownedLabel = "kubernetes-io-cluster-" + infraID
-		defer clusterHealthClient.Close()
-	} else if cluster.CloudProvider().ID() == "aws" {
-		clusterHealthClient, err = osdCloud.NewAwsCluster(ocmClient, o.clusterID, o.awsProfile)
+		for _, zone := range clusterHealthClient.GetAZs() {
+			instances, err := clusterHealthClient.GetAllVirtualMachines(zone)
+			if err != nil {
+				return err
+			}
+			for _, instance := range instances {
+				name := instance.Name
+				state := instance.State
+				labels := instance.Labels
+				belongsToCluster := false
+				for label := range labels {
+					if label == ownedLabel {
+						belongsToCluster = true
+					}
+				}
+				if !belongsToCluster {
+					if o.verbose {
+						log.Printf("Skipping a machine not belonging to the cluster: %s\n", name)
+					}
+					continue
+				}
+				totalCluster += 1
+				if state != "running" {
+					totalStopped += 1
+				} else {
+					if strings.HasPrefix(name, infraID) && strings.Contains(name, "master") {
+						runningMasters += 1
+					} else if strings.HasPrefix(name, infraID) && strings.Contains(name, "infra") {
+						runningInfra += 1
+					} else if strings.HasPrefix(name, infraID) && strings.Contains(name, "worker") {
+						runningWorkers += 1
+					}
+				}
+			}
+		}
+
+		healthObject.Actual.Stopped = totalStopped
+		healthObject.Actual.RunningMasters = runningMasters
+		healthObject.Actual.RunningInfra = runningInfra
+		healthObject.Actual.RunningWorker = runningWorkers
+		healthObject.Actual.Total = totalCluster
+
 		if err != nil {
+			log.Fatalf("Error getting instances %v", err)
 			return err
 		}
-		ownedLabel = "kubernetes.io/cluster/" + infraID
-		defer clusterHealthClient.Close()
+
+		healthOutput, err := yaml.Marshal(&healthObject)
+		if err != nil {
+			log.Fatalf("error: %v", err)
+		}
+		fmt.Printf("\n \n")
+		fmt.Println(string(healthOutput))
 	} else {
-		return errors.New(fmt.Sprintf("Unknown cloud provider found: %s", cluster.CloudProvider().ID()))
-	}
-	err = clusterHealthClient.Login()
-	if err != nil {
-		return err
-	}
-	for _, zone := range clusterHealthClient.GetAZs() {
-		instances, err := clusterHealthClient.GetAllVirtualMachines(zone)
+
+		hsHealthObject := createHypershiftHealthObject(cluster)
+
+		//var clusterHealthClient osdCloud.ClusterHealthClient
+
+		if cluster.Nodes().Compute() != 0 {
+			hsHealthObject.Expected.Worker = int(cluster.Nodes().Compute())
+		}
+		mc := ""
+		if cluster.Hypershift().Enabled() {
+			// Ignorning the error here as this endpoint is behind a specific permissioon
```

Comment:
What happens if there is an error? Shouldn't this command exit out since it will get an empty body back?